internal/config/schedule.go

```diff
@@ -72,38 +138,29 @@ func (r *RuntimeConfig) applyPendingSchedules() {
 		if pendingSchedule == nil {
 			delete(r.Schedules, id)
 		} else {
-			for _, memberRow := range pendingSchedule.MemberRows {
-				memberLogger := makeScheduleMemberLogger(r.logger.SugaredLogger, memberRow)
-
-				period := r.TimePeriods[memberRow.TimePeriodID]
-				if period == nil {
-					memberLogger.Warnw("ignoring schedule member for unknown timeperiod_id")
-					continue
-				}
-
-				var contact *recipient.Contact
-				if memberRow.ContactID.Valid {
-					contact = r.Contacts[memberRow.ContactID.Int64]
-					if contact == nil {
-						memberLogger.Warnw("ignoring schedule member for unknown contact_id")
-						continue
+			for _, rotation := range pendingSchedule.Rotations {
+
+				for _, member := range rotation.Members {
+					memberLogger := r.logger.With(
+						zap.Object("rotation", rotation),
+						zap.Object("rotation_member", member))
+
+					if member.ContactID.Valid {
+						member.Contact = r.Contacts[member.ContactID.Int64]
+						if member.Contact == nil {
+							memberLogger.Warnw("ignoring rotation member for unknown contact_id")
+							continue
```

Comment:
> Is the rotation member really ignored in that case?

Actually, it's not and the log message is misleading.

> Unless I'm overlooking something obvious, the member instance still remains in the `rotation.Members` map and will cause the entire `Schedule` being ignored in `debugVerifySchedule()`.

Ignored? That should trigger the following check and therefor trigger a panic:

https://github.com/Icinga/icinga-notifications/blob/0ff18c9e28afb82c3c810b292f2a384253dc2613/internal/config/verify.go#L212-L214
https://github.com/Icinga/icinga-notifications/blob/0ff18c9e28afb82c3c810b292f2a384253dc2613/internal/config/runtime.go#L76-L79

Something similar would also happen if both a contact and a group are set within the same rotation:

https://github.com/Icinga/icinga-notifications/blob/0ff18c9e28afb82c3c810b292f2a384253dc2613/internal/config/verify.go#L216-L218

This would violate the `CHECK (num_nonnulls(contact_id, contactgroup_id) = 1)` constraint, but similarly there should not be invalid IDs due to foreign key constraints, but after all, these are just additional sanity checks.

I think I'd just remove those checks from `debugVerifySchedule()`, the rest of the code will work perfectly fine with neither or both of `member.Contact` and `member.ContactGroup` set. Otherwise, this would just need more code to prevent a situation that should never happen and that wouldn't even cause trouble if it happened.